### PR TITLE
Bump pyam-iamc to >= 1.9.0

### LINKED ISSUE #27: Bump pyam-iamc to 1.8.0
Bumping up to the new version of pyam should yield a significant increase in speed when bigger files are read in.

See https://github.com/IAMconsortium/pyam/releases/tag/v1.8.0 (~3x times speed up for AR6 ensemble [from 90s down to 30s] )

### PR DESCRIPTION
**Pull request by phackstock**
closes #27.
- [ ] ~~Tests added~~
- [ ] ~~Documentation added~~
- [ ] ~~Example added (in the documentation, to an existing notebook, or in a new notebook)~~
- [x] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/iiasa/climate-assessment/pull/XX>`_) Added feature which does something``)

### CODE REVIEW

**Comment by phackstock** [comment]
Looks like there's four tests that are breaking. If any of you @znicholls, @lewisjared or @jkikstra could have a quick look that would be great.

**Comment by znicholls** [comment]
I think I worked it out. Put the code below after line 360 of `checks.py` and you'll see. I think the original regression is in pandas, but you can work around it by tweaking pyam

```python
        if model == "model7":
            import pdb
            pdb.set_trace()
            df_scen.timeseries()  # No CO2|Energy and Industrial as expected
            df_scen._data.index.get_level_values("variable").unique()  # No CO2|Energy and Industrial as expected

            # Using pyam accessors gives wrong result so `has_co2_energy`
            # evaluates incorrectly above
            df_scen.variable  # Contains CO2|Energy and Industrial ?!

            # What pyam uses internally. I think pyam is correct, but pandas
            # has a crazy regression where this returns incorrect information?!
            list(df_scen._data.index.levels[df_scen._data.index._get_level_number("variable")])  # Contains CO2|Energy and Industrial ?!
            # I assume that accessing levels directly on a MultiIndex is somehow not the intended pattern

            # I'd suggest updating pyam's internal's to use this, which seems to work
            list(df_scen._data.index.get_level_values("variable").unique())  # No CO2|Energy and Industrial as expected
```

File: src/climate_assessment/checks.py [hunk 1]
```diff
@@ -369,7 +369,7 @@ def check_reported_co2(df, filename, output_csv=False, outdir="output"):
         elif has_co2_total and has_co2_afolu:
             df_scen = _check_difference(df_scen, co2_total, co2_afolu, scen, model)
 
-        if co2_energy not in df_scen.variable:
+        if co2_energy not in df_scen._data.index.get_level_values("variable").unique():
```
Comment (znicholls) [inline src/climate_assessment/checks.py:372]: I think you'll need to make this replacement everywhere that `df_scen.variable` appears to get things to work properly.

**Review by znicholls — APPROVED** [review]
I think once the change is applied globally this should pass through. Option b is you do a bugfix release in pyam (fixing the pandas bug), do a new pyam release then change this MR and you don't have to do any fix at all here.

**Comment by phackstock** [comment]
> Option b is you do a bugfix release in pyam (fixing the pandas bug), do a new pyam release then change this MR and you don't have to do any fix at all here.

So you think it's a pandas bug? 
Not sure what's the best option here, if you can reproduce it, maybe open a pyam issue and we discuss over there? It's not super critical to update pyam for climate-assessment.
We could do #34 now then and just increase the pyam version later.

**Comment by danielhuppmann** [comment]
So I"m trying to get to the bottom of this, but I can't replicate the observation by @znicholls about `df.variable` returning something else than the correct values (with pyam 1.9 and pandas 1.5.3)... It seems all correct to me.

**Comment by danielhuppmann** [comment]
So I ran the tests along the entire history of pyam and it seems that https://github.com/IAMconsortium/pyam/pull/731 breaks the test.

**Comment by znicholls** [comment]
(I've made a bit of a mess of this, sorry)

- this CI is passing https://github.com/iiasa/climate-assessment/actions/runs/5752989542/job/15595215041 
- it's based on the branch in #41 
- that branch is the same as the branch in #42 
- so (I think), if you merge #42, this will pass (once black is applied). The fix is super yuck to look at, but that might be the 10 minute solution before working out the pandas headache if you need something quick

**Comment by danielhuppmann** [comment]
Implement a fix based on @znicholls' observation in https://github.com/IAMconsortium/pyam/issues/762

**Comment by phackstock** [comment]
After the pyam update, I wanted to update this PR to verify that that fixes the issue. However, as I just saw the tests were already passing yesterday? I'm a bit confused now but I guess it's all good?

**Comment by danielhuppmann** [comment]
See https://github.com/iiasa/climate-assessment/pull/36/commits/ccf0e7bba61f2526665ff196c62f442b0cb2fd19 - the problem was caused by the very inefficient groupby, so I refactored that piece of the code.

**Comment by phackstock** [comment]
Since the tests are passing and I'd love to get this out before the weekend, I'll move on with the merge. Hope that's ok @znicholls.

**Comment by znicholls** [comment]
> Hope that's ok @znicholls.

If all passing, all good (particularly given how much we've talked about this already)

**Pull request merged**